dashboard/app/views/levels/_free_response.html.haml

```diff
@@ -4,6 +4,8 @@
 
 - if in_level_group # The LevelGroup will collect results for each level.
   :javascript
+    window.levelGroup = window.levelGroup || {levels: {}};
+
     window.levelGroup.levels[#{level.id}] = {
```

Comment:
Do you think maybe this line should check whether window.levelGroup exists instead? That [happens twice](https://github.com/code-dot-org/code-dot-org/pull/9397/files#diff-e70bcfa464c70ffed4f3f11a7d95f5ceR34) later in this file, and it's consistent with the idea here - this template _might_ be loaded in a levelGroup context, we don't know.

More generally, over half of this template is JavaScript!  I thought we weren't doing this anymore!

![yuck_tigger](https://cloud.githubusercontent.com/assets/1615761/16698415/0098bad6-4503-11e6-90a0-e660bc7f3cf9.gif)